src/code/m_npc.c

```diff
@@ -0,0 +1,3975 @@
+            }
+        } else {
+            mNpc_GetNpcWorldNameP(npcName, npc->actor.fgName);
+        }
+        worldName = npcName;
+    }
+    mPr_CopyPlayerName(name, worldName);
+}
+
+void mNpc_GetRandomAnimalName(u8* dst) {
+    Animal_c* animal = common_data.animals;
+    u16 bitfield = 0;
+    s32 num = 0;
+    s32 select;
+    s32 i;
+
+    if (dst != NULL) {
+
+        for (i = 0; i < ANIMAL_NUM_MAX; i++) {
+            if (mNpc_CheckFreeAnimalInfo(animal) == FALSE) {
+                num++;
+                bitfield |= (1 << i);
+            }
+            animal++;
+        }
+
+        if (num > 0) {
+            animal = common_data.animals;
+            select = RANDOM(num);
+
+            for (i = 0; i < ANIMAL_NUM_MAX; i++) {
+                if (((bitfield >> i) & 1) == 1) {
+                    if (select == 0) {
+                        mNpc_GetNpcWorldNameAnm(dst, &animal->id);
+                        break;
+                    } else {
+                        select--;
+                    }
+                }
+                animal++;
+            }
+        }
+    }
+}
+
+void mNpc_GetAnimalPlateName(u8* dst, xyz_t wpos) {
+    UNUSED s32 pad;
+    Animal_c* animal = common_data.animals;
+    s32 bx;
+    s32 bz;
+    s32 utX;
+    s32 utZ;
+    s32 i;
+    Anmhome_c* home;
+
+    if (mFI_Wpos2BkandUtNuminBlock(&bx, &bz, &utX, &utZ, wpos) == TRUE) {
+
+        utX++;
+
+        for (i = 0; i < ANIMAL_NUM_MAX; i++) {
+            if (mNpc_CheckFreeAnimalInfo(animal) == FALSE) {
+                home = &animal->homeInfo;
+
+                if (home->blockX == bx && home->blockZ == bz && home->utX == utX && home->utZ == utZ) {
+                    mNpc_GetNpcWorldNameAnm(dst, &animal->id);
+                    break;
+                }
+            }
+
+            animal++;
+        }
+    }
+}
+
+s32 mNpc_GetNpcLooks(Actor* actor) {
+    s32 looks = NPC_LOOKS_BOY;
+
+    if (actor != NULL && actor->part == ACTOR_PART_NPC) {
+        Npc* npc = (Npc*)actor;
+
+        if (npc->npcInfo.animal != NULL) {
+            looks = npc->npcInfo.animal->id.looks;
+        }
+    }
+
+    return looks;
+}
+
+s32 mNpc_GetActorSex(u16 npcId) {
+    SpecialNpcData* spInfo = l_sp_actor_name;
+    s32 sex = NPC_SEX_MALE;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(l_sp_actor_name); i++) {
+        if (npcId == spInfo->specialNpcId) {
+            sex = spInfo->sex;
+            break;
+        }
+
+        spInfo++;
+    }
+
+    return sex;
+}
+
+s32 mNpc_GetLooks2Sex(s32 looks) {
+    s32 sex = NPC_SEX_OTHER;
+
+    if (looks == NPC_LOOKS_GIRL || looks == NPC_LOOKS_KO_GIRL || looks == NPC_LOOKS_NANIWA_LADY) {
+        sex = NPC_SEX_FEMALE;
+    } else if (looks == NPC_LOOKS_BOY || looks == NPC_LOOKS_SPORT_MAN || looks == NPC_LOOKS_GRIM_MAN) {
+        sex = NPC_SEX_MALE;
+    }
+
+    return sex;
+}
+
+s32 mNpc_GetAnimalSex(Animal_c* animal) {
+    s32 sex = NPC_SEX_OTHER;
```

Comment:
```suggestion
NpcSex mNpc_GetAnimalSex(Animal_c* animal) {
    NpcSex sex = NPC_SEX_OTHER;
```

Reply:
This gets called in a switch, and gives a warning that not all the cases (NPC_SEX_OTHER) are handled in the switch